### PR TITLE
[Parser] Parse br_table

### PR DESCRIPTION
**Pull request by tlively**

### CODE REVIEW

**Comment by tlively** [comment]
Current dependencies on/for this PR:
* main
  * **PR #6097** <a href="https://app.graphite.dev/github/pr/WebAssembly/binaryen/6097?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #6098** <a href="https://app.graphite.dev/github/pr/WebAssembly/binaryen/6098?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/WebAssembly/binaryen/6098?utm_source=stack-comment).

File: test/lit/wat-kitchen-sink.wast [hunk 1]
```diff
@@ -1607,6 +1607,115 @@
...
+    i32.const 0
+    br_table $a $b
+   end
+  end
+ )
+
+ ;; CHECK:      (func $br-table-multivalue (type $4) (result i32 i64)
+ ;; CHECK-NEXT:  (block $a (type $4) (result i32 i64)
+ ;; CHECK-NEXT:   (block $b (type $4) (result i32 i64)
+ ;; CHECK-NEXT:    (br_table $a $b
```
Comment (ashleynh) [inline test/lit/wat-kitchen-sink.wast:1697]: What's the proper way to read this?

File: src/parser/parsers.h [hunk 2]
```diff
@@ -1200,7 +1200,19 @@ template<typename Ctx> Result<> makeBreak(Ctx& ctx, Index pos) {
 }
 
 template<typename Ctx> Result<> makeBreakTable(Ctx& ctx, Index pos) {
-  return ctx.in.err("unimplemented instruction");
+  std::vector<typename Ctx::LabelIdxT> labels;
+  while (true) {
```
Comment (ashleynh) [inline src/parser/parsers.h:1204]: I'd add a comment here explaining the logic and the intended outcome of parsing at least one label.

**Review by ashleynh — APPROVED** [review]

**Comment by tlively** [comment]
### Merge activity

* **Nov 14, 7:40 PM**: @tlively started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/WebAssembly/binaryen/6098).

**Pull request merged**